From d25ea6e7a433b21e270491cdd420ebcc14016963 Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Tue, 16 May 2023 10:20:38 +0200
Subject: [PATCH 1/3] Revert "Linux: statx syscall number is always available"

This reverts commit 7255ccbec997312033c9fe48e0971a2e7cb19fbf.

Revert as this is not true for older kernels.
---
 sysdeps/unix/sysv/linux/statx.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/statx.c b/sysdeps/unix/sysv/linux/statx.c
index 8211ef4b68..9da6c0821c 100644
--- a/sysdeps/unix/sysv/linux/statx.c
+++ b/sysdeps/unix/sysv/linux/statx.c
@@ -25,14 +25,17 @@ int
 statx (int fd, const char *path, int flags,
        unsigned int mask, struct statx *buf)
 {
+#ifdef __NR_statx
   int ret = INLINE_SYSCALL_CALL (statx, fd, path, flags, mask, buf);
-#ifdef __ASSUME_STATX
+# ifdef __ASSUME_STATX
   return ret;
-#else
+# else
   if (ret == 0 || errno != ENOSYS)
     /* Preserve non-error/non-ENOSYS return values.  */
     return ret;
-  else
-    return statx_generic (fd, path, flags, mask, buf);
+# endif
+#endif
+#ifndef __ASSUME_STATX
+  return statx_generic (fd, path, flags, mask, buf);
 #endif
 }
-- 
2.41.0

